### ISSUE TITLE
[epsonprojector] "Unknown channel errmessage"

### ISSUE THREAD

**Issue by dicer** [openhab/openhab-addons#11703]
## Expected Behavior
I sometimes have a problem with the auto-iris on my projector. This results in an error being displayed on the "screen", but this error does not reach openhab.

## Current Behavior
The log contains the following, but the error msg itself does not reach the epsonErrMessage item, nor can I see anything about the auto-iris in the logs. Only an error code 10.

openhab.log:
```
[WARN ] [ternal.handler.EpsonProjectorHandler] - Unknown channel errmessage
```
This message repeats several times afterwards. It didn't show up anymore after a reboot of the projector.

events.log:
```
[INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'epsonprojector:projector-tcp:hometheater' changed from ONLINE to OFFLINE
[INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'epsonprojector:projector-tcp:hometheater' changed from OFFLINE to ONLINE
[INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'epsonPower' changed from OFF to UNDEF
[INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'epsonPowerState' changed from STANDBY to WARMUP
[INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'epsonPower' changed from UNDEF to ON
[INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'epsonSource' changed from 30 to UNDEF
[INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'epsonAspectRatio' changed from NORMAL to UNDEF
[INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'epsonColorMode' changed from LIVINGROOM to UNDEF
[INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'epsonMute' changed from OFF to UNDEF
[INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'epsonLuminance' changed from NORMAL to UNDEF
[INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'epsonBrightness' changed from 0 to UNDEF
[INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'epsonContrast' changed from 0 to UNDEF
[INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'epsonDensity' changed from 0 to UNDEF
[INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'epsonTint' changed from 0 to UNDEF
[INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'epsonColorTemperature' changed from 0 to UNDEF
[INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'epsonFleshTemperature' changed from 4 to UNDEF
[INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'epsonGamma' changed from G2_2 to UNDEF
[INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'epsonVerticalReverse' changed from ON to UNDEF
[INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'epsonHorizontalReverse' changed from OFF to UNDEF
[INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'epsonBackground' changed from BLUE to UNDEF
[INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'epsonErrCode' changed from 0 to 10
```

## Your Environment
openHAB 3.1.0
My projector is an Epson EH-TW3200 and I'm using the default thing/item/sitemap config from the documentation:
```
....
Number epsonErrCode     "Error Code [%d]"    <"siren-on">   { channel="epsonprojector:projector-tcp:hometheater:errcode" }
String epsonErrMessage  "Error Message [%s]" <"siren-off">  { channel="epsonprojector:projector-tcp:hometheater:errmessage" }
```

I can't really reproduce this as the error thankfully happens only once every couple of weeks.
Not really a big problem for me, so low priority, but I thought I'd report it anyway.

**Comment by mlobstein** [comment]
I think this is the same as issue #11126 in benqprojector. When trying to match the channel name to the channel enum in EpsonProjectorCommandType it was doing an equalsIgnoreCase(). This seems to be unreliable if the locale of the system uses a non-english language. The equalsIgnoreCase() was changed to equals() in PR #11141. You can try using a OH 3.2.0 snapshot build or just wait for the 3.2.0 release (which is due shortly) to see if the problem still exists there. In addition to this fix, the message in the Unknown Channel warn logging call was extended to show more details of the exception. So if the problem continues there will be more detail in the log to aid in tracing the problem

**Comment by dicer** [comment]
I'll wait and report back once I upgraded and the problem happens again. Thanks!

**Comment by dicer** [comment]
I updated to 3.2.0 yesterday and got the iris error just now:
```
==> events.log <==
2022-01-03 19:42:08.158 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'epsonErrCode' changed from NULL to 10

==> openhab.log <==
2022-01-03 19:42:08.253 [WARN ] [ternal.handler.EpsonProjectorHandler] - Unknown channel errmessage, exception: For input string: "0A"
```

**Comment by mlobstein** [comment]
I found the issue... It was using the error code hex value to lookup the ErrorMessage enum (which is keyed using int). So it has never worked if the error code is >= 10. PR to correct this will be opened.

**Comment by dicer** [comment]
I finally had the error *and* time to look at the data and everything works fine for me! Thanks for fixing this!
![Screenshot_20230224_203851](https://user-images.githubusercontent.com/219524/221275479-b5847fba-541e-4bd6-811f-da521b3d5a28.png)

**Comment by mlobstein** [comment]
> I finally had the error _and_ time to look at the data and everything works fine for me! Thanks for fixing this! ![Screenshot_20230224_203851](https://user-images.githubusercontent.com/219524/221275479-b5847fba-541e-4bd6-811f-da521b3d5a28.png)

Glad to hear that it works!
